backend/app/core/rate_limit.py

```
@@ -42,6 +46,106 @@ def _check_redis_available() -> bool:
+    if not settings.RATE_LIMIT_ENABLED:
+        return None
+    if _redis_rate_limit_client is not None:
+        return _redis_rate_limit_client
+    try:
+        import redis
+
+        client = redis.from_url(settings.REDIS_URL, socket_connect_timeout=1)
+        client.ping()
+        _redis_rate_limit_client = client
+        return client
+    except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)